src/repository/repository.class.ts

```diff
@@ -0,0 +1,45 @@
+import { DataSource, EntitySchema, Repository as ORMRepository } from 'typeorm';
+
+export class Repository<TEntity> {
+  private repository: ORMRepository<EntitySchema<TEntity>>;
+
+  getAllItems: () => Promise<EntitySchema<TEntity>[]>;
+
+  getItemById: (_id: string) => Promise<EntitySchema<TEntity> | null>;
+
+  getItemByWhere: (_where: object) => Promise<EntitySchema<TEntity> | null>;
+
+  createItem: (_item: EntitySchema<TEntity>) => Promise<EntitySchema<TEntity>>;
+
+  updateItem: (
```

Comment:
Still have to update these to updateItemById and so on

Reply:
done.